update-go-version.sh

```diff
@@ -12,14 +12,17 @@ echo "Updating golang version to $golang_base_version"
 go mod edit -go ${golang_base_version}
 go mod edit -go ${golang_base_version} tools/go.mod
 sed -i "s,^GOVERSION = 1.[0-9]\+,GOVERSION = ${golang_base_version}," Makefile
-sed -i "s,^\(FROM registry.ci.openshift.org/openshift/release:rhel-8-release-golang-\)1.[0-9]\+,\1${golang_base_version}," images/*/Dockerfile
-sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registry.access.redhat.com/ubi8/go-toolset:${golang_base_version}," images/*/Dockerfile
-sed -i "s,^FROM registry.access.redhat.com/ubi8/go-toolset:[.0-9]\+,FROM registry.access.redhat.com/ubi8/go-toolset:${golang_base_version}," images/*/Containerfile
+
+golang_major_version=$(echo ${golang_base_version} | cut -d '.' -f 1,2)
```

Comment:
The comment at the beginning of the file will need to be updated
```
# Updates our various build files, CI configs, ... to the latest released go
# version in a given minor release stream (1.x)
```
as you change the format of the parameter which needs to be passed to this script, and what the script used to call "minor release stream" is now called "golang major version" (which is more accurate https://go.dev/doc/devel/release talks about major releases (1.21.0) and minor revisions (1.21.4))

An alternative to these changes would be to keep passing `1.21` as the argument to the script, and to only change the `go.mod` modification to append a `.0` to the argument. `./update-go-version 1.21.0` is misleading as most of the changes will be about using the latest 1.21.x version available, not about using 1.21.0 precisely.

Reply:
>An alternative to these changes would be to keep passing 1.21 as the argument to the script, and to only change the go.mod modification to append a .0 to the argument. ./update-go-version 1.21.0 is misleading as most of the changes will be about using the latest 1.21.x version available, not about using 1.21.0 precisely.

yes, this is much better because only go mod file need to have that toolchain update but runner and images doesn't